### PR TITLE
Prevent us from saving insane tx combinations into the db

### PR DESCRIPTION
**Pull request by luckysori**
By adding this we can detect problems _before_ we commit to deleting all the corresponding information in `cfds` and `events`, meaning that we can more easily debug problems related to the insane-tx-combinations error.

As an example, debugging https://github.com/itchysats/itchysats/issues/1842 would have been easier.

### CODE REVIEW

File: daemon/src/db.rs [hunk 1]
```diff
@@ -424,22 +424,54 @@ impl Connection {
...
+                        insert_refund_tx(&mut db_tx, id, refund).await?;
+                    }
+                    ClosedCfdInput {
+                        collaborative_settlement,
+                        commit,
+                        non_collaborative_settlement,
+                        refund,
+                        ..
+                    } => bail!(
+                        "CFD to be closed has insane combination of transactions:
```
Comment (bonomat) [inline daemon/src/db.rs:469]: I can see this error already popping up in our logs 🙈

**Review by bonomat — APPROVED** [review]
LGTM

File: daemon/src/db.rs (hunk 1 above)
Comment (luckysori) [reply daemon/src/db.rs:469]: Hahah yes! Better earlier than later, as @da-kami pointed out offline.

**Pull request merged**